pkg/trace/operation.go

```diff
@@ -208,6 +261,11 @@ func FromOperation(parent Operation, format string, args ...interface{}) Operati
 //   A new operation
 func FromContext(ctx context.Context, message string) Operation {
 
+	// do we have an operation
+	if op, ok := ctx.(Operation); ok {
+		return op
+	}
+
 	// do we have a context w/the op added as a value
 	if op, ok := ctx.Value(OpTraceKey).(operation); ok {
 		// ensure we have an initialized operation
```

Comment:
`if op == nil || op.id == ""`

Reply:
In this case we have an `operation` (not a `*operation`), so I don't think it can be `nil`.